### ISSUE TITLE
Linter reports sloppyTypeAssert for slice-type assertions (VS Code)

### ISSUE THREAD

**Issue by sammy-hughes** [go-critic/go-critic#1422]
The go-critic plugin for VS Code incorrectly highlights type-assertions with sloppyTypeAssert if the type assertion statement asserts a slice-type. This was initially observed for a declared type as a slice of a struct type. Changing the statement does not change whether go-critic warns on such a statement. The statement was first changed to assert the slice-of-declared-type and then to assert to slice-of-string.

Example lintings erroneous:
```Go
type Example struct {A string}
type Examples []Example

func Input() any {
    return nil
}

func main() {
    data := Input()
    
    exampleA, ok := data.(Examples)
    fmt.Println(ok, exampleA)
    
    exampleB, ok := data.([]Example)
    fmt.Println(ok, exampleB)
    
    exampleC, ok := data.([]string)
    fmt.Println(ok, exampleC)
}
```

The above are each (independently) valid assertions, but go-critic reports `sloppyTypeAssert:  type assertion from/to types are identical`.

**Comment by quasilyte** [comment]
Looks like a legit bug, thanks for reporting.
